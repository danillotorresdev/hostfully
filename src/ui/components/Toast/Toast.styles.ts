import { styled } from "@linaria/react";

export const ToastContainer = styled.div<{
  type?: "success" | "error" | "warning";
}>`
  position: fixed;
  bottom: 20px;
  right: 20px;
  padding: 10px 20px;
  z-index: 9999;
  width: 330px;
  height: 35px;
  font-size: 1.2rem;
  display: flex;
  border-radius: 5px;
  align-items: center;
  background-color: ${({ type }) => {
    switch (type) {
      case "success":
        return "#4caf50";
      case "error":
        return "#f44336";
      case "warning":
        return "#ff9800";
      default:
        return "#333";
    }
  }};
  color: #fff;
  transition: transform 0.5s ease-in-out;
`;
