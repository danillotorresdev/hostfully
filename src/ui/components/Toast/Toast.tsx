import { createPortal } from "react-dom";
import * as S from "@/ui/components/Toast/Toast.styles";

const portalRoot = document.getElementById("portal-root") as HTMLElement;

export type ToastProps = {
  message: string;
  type?: "success" | "error" | "warning";
};

export const ToastComponent = ({ message, type }: ToastProps) => (
  <S.ToastContainer type={type}>{message}</S.ToastContainer>
);

export const Toast = ({ message, type }: ToastProps) =>
  createPortal(<ToastComponent message={message} type={type} />, portalRoot);
