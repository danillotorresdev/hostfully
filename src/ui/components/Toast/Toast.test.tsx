import { ToastComponent } from "./Toast";
import { beforeEach, describe, expect, it, vi } from "vitest";

import { render } from "@/utils/settings/tests/utilities";

describe("Toast", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("should render the component", () => {
    const { container } = render(
      <ToastComponent message="This is a message" type="success" />,
    );

    expect(container).toMatchSnapshot();
  });
});
