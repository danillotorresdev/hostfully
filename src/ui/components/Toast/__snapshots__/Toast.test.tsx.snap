// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast > should render the component 1`] = `
<div>
  <div
    class="mocked-styled-0 tjr5k2h"
    style="--tjr5k2h-0: #4caf50;"
    type="success"
  >
    This is a message
  </div>
</div>
`;
